great_expectations/datasource/fluent/config.py

```diff
@@ -221,6 +228,8 @@ def _load_datasource_subtype(cls, v: List[dict]):
                 # attach the datasource to the nested assets, avoiding recursion errors
                 for asset in datasource.assets:
                     asset._datasource = datasource
+                    for batch_config in asset.batch_configs:
+                        batch_config._data_asset = asset
```

Comment:
Could we use a setter instead of touching a private attr?

Reply:
I would prefer that too, but my understanding is that there is a bug in pydatic that prevents that: https://github.com/pydantic/pydantic/issues/1577.